flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtilsTest.java

```diff
@@ -0,0 +1,323 @@
+		conf.setFloat(TaskManagerOptions.MANAGED_MEMORY_OFFHEAP_FRACTION, fraction);
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> {
+			assertThat(taskExecutorResourceSpec.getOffHeapManagedMemorySize(), is(taskExecutorResourceSpec.getManagedMemorySize().multiply(fraction)));
+			assertThat(taskExecutorResourceSpec.getOnHeapManagedMemorySize(), is(taskExecutorResourceSpec.getManagedMemorySize().subtract(taskExecutorResourceSpec.getOffHeapManagedMemorySize())));
+		});
+	}
+
+	@Test
+	public void testConfigJvmMetaspaceSize() throws Exception {
+		final MemorySize jvmMetaspaceSize = MemorySize.parse("50m");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_METASPACE, jvmMetaspaceSize.getMebiBytes() + "m");
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> assertThat(taskExecutorResourceSpec.getJvmMetaspaceSize(), is(jvmMetaspaceSize)));
+	}
+
+	@Test
+	public void testConfigJvmOverheadSizeAbsolute() throws Exception {
+		final MemorySize jvmOverheadSize = MemorySize.parse("50m");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MAX, jvmOverheadSize.getMebiBytes() + "m");
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MIN, jvmOverheadSize.getMebiBytes() + "m");
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> assertThat(taskExecutorResourceSpec.getJvmOverheadSize(), is(jvmOverheadSize)));
+	}
+
+	@Test
+	public void testConfigJvmOverheadSizeFraction() throws Exception {
+		final MemorySize minSize = MemorySize.parse("50m");
+		final MemorySize maxSize = MemorySize.parse("200m");
+		final float fraction = 0.2f;
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MAX, maxSize.getMebiBytes() + "m");
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MIN, minSize.getMebiBytes() + "m");
+		conf.setFloat(TaskManagerOptions.JVM_OVERHEAD_FRACTION, fraction);
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> {
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize().getBytes(), greaterThanOrEqualTo(minSize.getBytes()));
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize().getBytes(), lessThanOrEqualTo(maxSize.getBytes()));
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize(), anyOf(
+				is(minSize),
+				is(maxSize),
+				is(taskExecutorResourceSpec.getTotalProcessMemorySize().multiply(fraction))
+			));
+		});
+	}
+
+	@Test
+	public void testConfigTotalFlinkMemorySize() throws Exception {
+		final MemorySize totalFlinkMemorySize = MemorySize.parse("1g");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.TOTAL_FLINK_MEMORY, totalFlinkMemorySize.getMebiBytes() + "m");
+
+		TaskExecutorResourceSpec taskExecutorResourceSpec = TaskExecutorResourceUtils.resourceSpecFromConfig(conf);
+		assertThat(taskExecutorResourceSpec.getTotalFlinkMemorySize(), is(totalFlinkMemorySize));
```

Comment:
Shouldn't we also assert that we have configured the remaining values correctly?

Reply:
`TaskExecutorResourceSpec#getTotalFlinkMemorySize` gets the Total Flink Memory size by adding up individual memory sizes. So this test case only need to validate that the added up Total Flink Memory size equals to the configured value.